src/conn/conn_cache.c

```diff
@@ -8,6 +8,38 @@
 
 #include "wt_internal.h"
 
+/*
+ * __cache_config_abs_to_pct --
+ *	Convert the absolute value in cache config to percentage.
+ */
+static inline int
+__cache_config_abs_to_pct(WT_SESSION_IMPL *session,
+    double *param, const char *param_name)
+{
+	WT_CONNECTION_IMPL *conn;
+
+	conn = S2C(session);
+
+	WT_ASSERT(session, param != NULL);
+
+	/*
+	 * Anything above 100 is an absolute value; convert it to percentage.
+	 * An absolute value can't exceed the cache size.
+	 */
+	if (*param > 100.0) {
+		if (*param > conn->cache_size)
+			WT_RET_MSG(session, EINVAL,
+			    "%s should not exceed cache size", param_name);
+		/*
+		 * Avoid division by zero if the cache size has not yet been set
```

Comment:
Once you take my advice about disallowing absolute values and shared cache it should be possible to replace this with an assertion that the shared cache isn't configured (unless there is an ordering problem).